shell/common/shell_unittests.cc

```diff
@@ -2824,6 +2857,113 @@ TEST_F(ShellTest, Spawn) {
   ASSERT_FALSE(DartVMRef::IsInstanceRunning());
 }
 
+TEST_F(ShellTest, SpawnWithDartEntrypointArgs) {
+  auto settings = CreateSettingsForFixture();
+  auto shell = CreateShell(settings);
+  ASSERT_TRUE(ValidateShell(shell.get()));
+
+  auto configuration = RunConfiguration::InferFromSettings(settings);
+  ASSERT_TRUE(configuration.IsValid());
+  configuration.SetEntrypoint("canRecieveArgumentsWhenEngineRun");
+  const std::vector<std::string> entrypoint_args{"foo", "bar"};
```

Comment:
Shouldn't there be asserts in this test that these are getting passed in on the ui thread?

Reply:
After the arguments are passed to dart, they will be verified on the dart side, https://github.com/flutter/engine/blob/6a5d40fef2c61b9265de32ef771093cfa8e5efa5/shell/common/fixtures/shell_test.dart#L257, and then assert here ,
https://github.com/flutter/engine/blob/6a5d40fef2c61b9265de32ef771093cfa8e5efa5/shell/common/shell_unittests.cc#L2884 .
This test case refers to this , 
https://github.com/flutter/engine/blob/5a26133f9eb5968774c38460783f30ec4b842c38/runtime/dart_isolate_unittests.cc#L335

Reply:
SGTM, missed that.  I wish these things weren't so far apart but nothing we can do about that.